src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/incidents/IncidentsSynchronisationService.kt

```diff
@@ -37,32 +39,70 @@ class IncidentsSynchronisationService(
       )
       return
     }
-
-    val nomisIncident = nomisApiService.getIncident(event.incidentCaseId)
-    incidentsMappingService.findByNomisId(nomisIncidentId = event.incidentCaseId)
+    val nomisIncidentId = event.incidentCaseId
+    val nomisIncident = nomisApiService.getIncident(nomisIncidentId)
+    incidentsMappingService.findByNomisId(nomisIncidentId = nomisIncidentId)
       ?.let {
         // Should never happen - it shouldn't exist in the mapping table
         telemetryClient.trackEvent(
           "incidents-synchronisation-created-ignored",
           event.toTelemetryProperties(),
         )
-      } ?: let {
-      incidentsService.upsertIncident(
-        NomisSyncRequest(
-          id = null,
-          initialMigration = false,
-          incidentReport = nomisIncident.toNomisIncidentReport(),
-        ),
-      ).also { dpsIncident ->
-        tryToCreateIncidentMapping(event, dpsIncident.id.toString()).also { result ->
-          telemetryClient.trackEvent(
-            "incidents-synchronisation-created-success",
-            event.toTelemetryProperties(
-              dpsIncident.id.toString(),
-              result == MAPPING_FAILED,
-            ),
-          )
+      }
+      ?: try {
+        incidentsService.upsertIncident(
+          NomisSyncRequest(
+            id = null,
+            initialMigration = false,
+            incidentReport = nomisIncident.toNomisIncidentReport(),
+          ),
+        ).also { dpsIncident ->
+          tryToCreateIncidentMapping(event, dpsIncident.id.toString()).also { result ->
+            telemetryClient.trackEvent(
+              "incidents-synchronisation-created-success",
+              event.toTelemetryProperties(
+                dpsIncident.id.toString(),
+                result == MAPPING_FAILED,
+              ),
+            )
+          }
         }
+      } catch (e: WebClientResponseException.Conflict) {
+        // We have a conflict - this should only ever happen if the incident was stored in DPS, but we didn't receive a response in time
+        // so is never added to the incidents mapping table
+        log.error("Conflict received from DPS for nomisIncidentId: $nomisIncidentId, attempting to recover.", e)
```

Comment:
not sure this is an error? More of a warning perhaps.  Do you need to log `e` as well or just `e.message`?

Reply:
maybe even information since it is handled

Reply:
yeah you're right - I'll do mini update